iceoryx_posh/test/mocks/roudi_memory_provider_mock.hpp

```diff
@@ -42,6 +42,11 @@ class MemoryProviderTestImpl : public iox::roudi::MemoryProvider
 
 #if defined(QNX) || defined(QNX__) || defined(__QNX__)
         dummyMemory = static_cast<uint8_t*>(memalign(alignment, size));
+#elif defined(__APPLE__)
+        uint64_t memOffset = reinterpret_cast<uint64_t>(malloc(size + alignment));
+        dummyMemory = reinterpret_cast<uint8_t*>(memOffset);
+        memOffset += memOffset % alignment;
```

Comment:
this has bugs, please use the proposal from chunck_mock.hpp

Reply:
done